tests/santa/test_api_views.py

```diff
@@ -2158,3 +2229,216 @@ def test_delete_enrollment_not_found(self):
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.json(), {'count': 0, 'next': None, 'previous': None, 'results': []})
+
+    def test_list_targets_filter_by_target_type(self):
+        self.set_permissions("santa.view_target")
+        response = self.get(reverse('santa_api:targets'), {"target_type": "BINARY"})
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.json(), {
+            'count': 1,
+            'next': None,
+            'previous': None,
+            'results': [
+                {
+                    'id': self.file_target.pk,
+                    'files': [
+                        {
+                            'id': self.file_target.files[0].id,
+                            'name': self.file_target.files[0].name,
+                            'path': '/Library/Frameworks/Compressor.framework/Versions/A/'
+                                    'Resources/CompressorTranscoderX.bundle/Contents/MacOS',
+                            'sha_256': self.file_sha256,
+                            'bundle_path': '/Library/Frameworks/Compressor.framework/Versions/A/'
+                                           'Resources/CompressorTranscoderX.bundle',
+                            'source': self.file_target.files[0].source.pk,
+                            'bundle': self.file_target.files[0].bundle.pk,
+                            'signed_by': self.file_target.files[0].signed_by.pk,
+                        }
+                    ],
+                    'certificates': [],
+                    'team_ids': [],
+                    'type': 'BINARY',
+                    'identifier': self.file_sha256
+                }
+            ]})
+
+    def test_list_targets_filter_by_identifier(self):
+        self.set_permissions("santa.view_target")
+        response = self.get(reverse('santa_api:targets'), {"target_identifier": self.file_team_id})
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.json(), {
+            'count': 1,
+            'next': None,
+            'previous': None,
+            'results': [
+                {
+                    'id': self.file_team_id_target.pk,
+                    'files': [],
+                    'certificates': [],
+                    'team_ids': [
+                        {
+                            'organizational_unit': self.file_team_id,
+                            'organization': 'Apple Inc.'
+                        }
+                    ],
+                    'type': 'TEAMID',
+                    'identifier': self.file_team_id
+                }
+            ]})
+
+    def test_list_targets_pagination(self):
+        for _ in range(0, 10):
+            self.force_rule()
+        self.set_permissions("santa.view_target")
+        response = self.get(reverse('santa_api:targets'), {"page_size": 5})
```

Comment:
You could reduce the page_size even more to have less targets to create.

Reply:
done